xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/UserMentionsFormatter.java

```diff
@@ -66,7 +69,7 @@ public String formatMention(String userReference, DisplayStyle style)
         if (Objects.equals(style, LOGIN) || Objects.equals(firstName, "") && Objects.equals(lastName, "")) {
             // if the login is asked explicitly we display it
             // if the user has no first name and no last name, we display the login by default
-            content = this.documentReferenceResolver.resolve(userReference).getName();
+            content = this.documentReferenceResolver.resolve(actorReference).getName();
```

Comment:
That would need a todo for later: I guess you're doing it like that because the User API doesn't allow to retrieve the login of a user, right? But right now it will work only because we know that any user reference we obtain is also a document reference.

Reply:
TODO added

Reply:
Ok maybe create a JIRA for it too so we don't forget

Reply:
https://jira.xwiki.org/browse/XWIKI-17999